chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -382,198 +418,207 @@
+
+      template "#{os_dir}/net-post-install.sh" do
+        mode 0644
+        owner "root"
+        group "root"
+        variables(admin_web: admin_web,
+                  os_codename: os_codename,
+                  repos: node[:provisioner][:repositories][os][arch],
+                  admin_ip: admin_ip,
+                  provisioner_web: provisioner_web,
+                  web_path: web_path)
+      end
 
-    template "#{os_dir}/crowbar_join.sh" do
-      mode 0644
-      owner "root"
-      group "root"
-      source "crowbar_join.ubuntu.sh.erb"
-      variables(admin_web: admin_web,
-                os_codename: os_codename,
-                crowbar_repo_web: crowbar_repo_web,
-                admin_ip: admin_ip,
-                provisioner_web: provisioner_web,
-                web_path: web_path)
-    end
+      template "#{os_dir}/crowbar_join.sh" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "crowbar_join.ubuntu.sh.erb"
+        variables(admin_web: admin_web,
+                  os_codename: os_codename,
+                  crowbar_repo_web: crowbar_repo_web,
+                  admin_ip: admin_ip,
+                  provisioner_web: provisioner_web,
+                  web_path: web_path)
+      end
 
-  when /^(hyperv|windows)/ =~ os
-    template "#{tftproot}/adk-tools/build_winpe_#{os}.ps1" do
-      mode 0644
-      owner "root"
-      group "root"
-      source "build_winpe_os.ps1.erb"
-      variables(os: os,
-                admin_ip: admin_ip)
-    end
+    when /^(hyperv|windows)/ =~ os
+      # Windows is x86_64-only
+      os_dir = "#{tftproot}/#{os}"
+
+      template "#{tftproot}/adk-tools/build_winpe_#{os}.ps1" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "build_winpe_os.ps1.erb"
+        variables(os: os,
+                  admin_ip: admin_ip)
+      end
 
-    directory "#{os_dir}" do
-      mode 0755
-      owner "root"
-      group "root"
-      action :create
-    end
+      directory "#{os_dir}" do
+        mode 0755
+        owner "root"
+        group "root"
+        action :create
+      end
 
-    # Let's stay compatible with the old code and remove the per-version extra directory
-    if File.directory? "#{os_dir}/extra"
-      directory "#{os_dir}/extra" do
-        recursive true
-        action :delete
+      # Let's stay compatible with the old code and remove the per-version extra directory
+      if File.directory? "#{os_dir}/extra"
+        directory "#{os_dir}/extra" do
+          recursive true
+          action :delete
+        end
       end
-    end
 
-    link "#{os_dir}/extra" do
-      action :create
-      to "../windows-common/extra"
-    end
+      link "#{os_dir}/extra" do
+        action :create
+        to "../windows-common/extra"
+      end
 
-    missing_files = ! File.exists?("#{os_dir}/boot/bootmgr.exe")
-  end
+      missing_files = !File.exists?("#{os_dir}/boot/bootmgr.exe")
```

Comment:
Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.